naga/src/valid/expression.rs

```diff
@@ -659,11 +659,10 @@ impl super::Validator {
 
                 match (sample, class.is_multisampled()) {
                     (None, false) => {}
-                    (Some(sample), true) => {
-                        if resolver[sample].scalar_kind() != Some(Sk::Sint) {
-                            return Err(ExpressionError::InvalidImageOtherIndexType(sample));
-                        }
-                    }
+                    (Some(sample), true) => match resolver[sample] {
+                        Ti::Scalar(Sc::I32 | Sc::U32) => {}
```

Comment:
nitpick: This sort of `if`-ish control flow is exactly why `matches!(…)` exists. Can we use that instead?